### PR TITLE
feat: extended the default step lock handling to allow disabling it completly

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
* Added const strings which allow to reference parameter names
* Added new paramter 'EnableLockHandling' which allows disabling the whole locking mechanic

### CODE REVIEW

File: Runtime/RestrictiveEnvironment/DefaultStepLockHandling.cs [hunk 1]
```diff
@@ -11,12 +11,33 @@ namespace Innoactive.Creator.Core.RestrictiveEnvironment
...
+
+        /// <summary>
+        /// To stop locking every LockableProperty at the beginning of the training set this parameter to false.
+        /// </summary>
+        public const string LockCourseOnStart = "LockCourseOnStart";
+
+        /// <summary>
+        /// To lock every LockableProperty at the end of the training set this parameter to true.
+        /// </summary>
+        public const string LockOnCourseFinished = "LockOnCourseFinished";
```
Comment (Gusinuhe) [inline Runtime/RestrictiveEnvironment/DefaultStepLockHandling.cs:27]: These are strings and constants, they can't be set to true or false.

File: Runtime/RestrictiveEnvironment/DefaultStepLockHandling.cs [hunk 2]
```diff
@@ -11,13 +11,34 @@ namespace Innoactive.Creator.Core.RestrictiveEnvironment
...
     {
+        /// <summary>
+        /// If you want to disable lock handling set this parameter to false.
+        /// </summary>
+        public const string EnableLockHandlingParameterName = "EnableLockHandling";
+
+        /// <summary>
+        /// To stop locking every LockableProperty at the beginning of the training set this parameter to false.
+        /// </summary>
+        public const string LockCourseOnStartParameterName = "LockCourseOnStart";
```
Comment (Luphoxa) [inline Runtime/RestrictiveEnvironment/DefaultStepLockHandling.cs:22]: Here it's called LockCourse**On**Start and in line 29 it's called **lockOnCourseStart**
And the other parameter is called lock**On**Course finished. It's a little confusing.
In #118 it is now written down as LockOnCourseStart as well, so I would suggest using that naming then.

File: Runtime/RestrictiveEnvironment/DefaultStepLockHandling.cs [hunk 3]
```diff
@@ -11,12 +11,33 @@ namespace Innoactive.Creator.Core.RestrictiveEnvironment
...
+
+        /// <summary>
+        /// To lock every LockableProperty at the end of the training set this parameter to true.
+        /// </summary>
+        public const string LockOnCourseFinishedParameterName = "LockOnCourseFinished";
```
Comment (aleksei-korolev) [inline Runtime/RestrictiveEnvironment/DefaultStepLockHandling.cs:27]: Inconsisted naming:

1. It should be LockOnCourse...
2. It should be either LockOnCourseStart and LockOnCourseFinish, or LockOnCourseStarted nad LockOnCourseFinished.

Personally, I would prefer LockWhenCourseStarts and LockWhenCourseFinishes

**Comment by SimonTheSourcerer** [comment]
Disabling the locking while the course is running is much more complex than i thought it is. So this actually does not work properly so I will close this PR.

**Pull request closed without merging**